intake_accumulo/source.py

```diff
@@ -0,0 +1,71 @@
+from collections import namedtuple, OrderedDict
+
+from intake.source import base
+
+import pandas as pd
+
+dtypes = [('row', 'str'),
+          ('column_family', 'str'),
+          ('column_qualifier', 'str'),
+          ('column_visibility', 'str'),
+          ('time', 'datetime64[ms]'),
+          ('value', 'object')]
+
+KeyValue = namedtuple('KeyValue', [dtype[0] for dtype in dtypes])
+
+
+class AccumuloSource(base.DataSource):
+    """Read data from Accumulo table.
+
+    Parameters
+    ----------
+    table : str
+        The database table that will act as source
+    host : str
+        The server hostname for the given table
+    port : int
+        The server port for the given table
+    username : str
+        The username used to connect to the Accumulo cluster
+    password : str
+        The password used to connect to the Accumulo cluster
+    """
+
+    def __init__(self, table, host, port, username, password, metadata=None):
+        from .accumulo import Accumulo
+
+        self._client = Accumulo(host, port, username, password)
+        self._table = table
+
+        super(AccumuloSource, self).__init__(container='dataframe',
+                                             metadata=metadata)
+
+    def _get_schema(self):
+        return base.Schema(datashape=None,
+                           dtype=OrderedDict(dtypes),
```

Comment:
dtypes is always the same? We often use a zero-length dataframe for this function, though in this case it may not matter as the index is uninteresting.

Reply:
Yes, the dtypes is always the same. Accumulo tables have a fixed key-value schema.

I replaced the dtype parameter with `dtype=None`, but the tests failed. Was it a typical `dtype` value?

Reply:
It would be a zero-length data-frame, with the same dtypes, like
```
pd.DataFrame({c: pd.Series([], dtype=d) for (c, d) in dtypes})
```